src/mlpack/bindings/python/tests/test_python_binding.py

```diff
@@ -713,6 +749,308 @@ def testModel(self):
+
+  def testTwoDimensionCol(self):
+    """
+    Test that we pass Two Dimension column vetor as input paramter
+    """
+    x = np.random.rand(100,1)
+    z = copy.copy(x)
+
+    output = test_python_binding(string_in='hello',
+                                 int_in=12,
+                                 double_in=4.0,
+                                 col_in=z)
+
+    self.assertEqual(output['col_out'].shape[0], 100)
+    self.assertEqual(output['col_out'].dtype, np.double)
+
+    for i in range(100):
+      self.assertEqual(output['col_out'][i], x[i] * 2)
+
+  def testTwoDimensionColForceCopy(self):
+    """
+    Test that we pass Two Dimension column vetor as input paramter
+    """
+    x = np.random.rand(100,1)
+
+    output = test_python_binding(string_in='hello',
+                                 int_in=12,
+                                 double_in=4.0,
+                                 col_in=x,
+                                 copy_all_inputs=True)
+
+    self.assertEqual(output['col_out'].shape[0], 100)
+    self.assertEqual(output['col_out'].dtype, np.double)
+
+    for i in range(100):
+      self.assertEqual(output['col_out'][i], x[i] * 2)
+
+  def testTwoDimensionUcol(self):
+    """
+    Test that we pass Two Dimension unsigned column vector input parameter.
+    """
+    x = np.random.randint(0, high=500, size=(100,1))
+    z = copy.copy(x)
+
+    output = test_python_binding(string_in='hello',
+                                 int_in=12,
+                                 double_in=4.0,
+                                 ucol_in=z)
+
+    self.assertEqual(output['ucol_out'].shape[0], 100)
+    self.assertEqual(output['ucol_out'].dtype, np.long)
+    for i in range(100):
+      self.assertEqual(output['ucol_out'][i], x[i] * 2)
+
+  def testTwoDimensionUcolForceCopy(self):
+    """
+    Test that we pass Two Dimension unsigned column vector input parameter.
+    """
+    x = np.random.randint(0, high=500, size=(100,1))
+
+    output = test_python_binding(string_in='hello',
+                                 int_in=12,
+                                 double_in=4.0,
+                                 ucol_in=x,
+                                 copy_all_inputs=True)
+
+    self.assertEqual(output['ucol_out'].shape[0], 100)
+    self.assertEqual(output['ucol_out'].dtype, np.long)
+    for i in range(100):
+      self.assertEqual(output['ucol_out'][i], x[i] * 2)
+
+  def testOneDimensionMatrixAndInfoPandas(self):
+    """
+    Test that we can pass a one dimension matrix with some categorical features.
+    """
+    x = pd.DataFrame(np.random.rand(10), columns=list('a'))
+    x['e'] = pd.Series(['a', 'b', 'c', 'd', 'a', 'b', 'e', 'c', 'a', 'b'],
+        dtype='category')
+    z = copy.copy(x)
+
+    output = test_python_binding(string_in='hello',
+                                 int_in=12,
+                                 double_in=4.0,
+                                 matrix_and_info_in=z)
```

Comment:
Hmm, technically I think the idea would be that we pass `x['e']` directly here?

Reply:
Here I pass `x ` i.e. 1-D and it is without column from `matrix_and_info_in`

Reply:
Well, now I'm a little bit confused.  I thought that this test was testing that passing a Pandas series worked, but it seems like you're just passing a 2-D Pandas dataframe with one column.

Reply:
Before I was testing that if we can pass 1-D matrix after casting it  to 2-D (as `matrix_and_info_in_reshape`) from  `arma_numpy.numpy_to_mat_d()`

Reply:
Ah, okay, but I don't think that what we're passing is actually a 1-d Pandas object.  For that we'd need to pass a `Series` object, so I guess we'd need to do `z[0]` here in the call for `matrix_and_info_in`.

Reply:
By the way, everything else in the PR looks good to me, so I'm ready to approve it once we handle this last comment.